torch/optim/optimizer.py

```diff
@@ -51,6 +52,29 @@ def __init__(self, params, defaults):
         for param_group in param_groups:
             self.add_param_group(param_group)
 
+    def __new__(cls, *args, **kwargs):
+        self = super(Optimizer, cls).__new__(cls)
+
+        def hook(func_to_hook, profile_type):
+            profile_name = "{}#{}.{}".format(profile_type, self.__class__.__name__, func_to_hook.__name__)
```

Comment:
it seems the hook function's responsibility is only construct the profile name? if so, we can remove it.